htlcswitch/link.go

```diff
@@ -2419,6 +2472,49 @@ func (l *channelLink) handleUpstreamMsg(msg lnwire.Message) {
 
 }
 
+// drivePendingStfuSends checks to see if we owe any Stfu messages to our peer.
+// If we do then we will send them and record that send. We can owe our peer an
+// Stfu either if we have a pending request to quiesce the channel or our peer
+// has sent us an Stfu to which we haven't yet responded.
+func (l *channelLink) drivePendingStfuSends() {
+	if !l.quiescer.oweStfu() {
+		return
+	}
+
+	l.log.Debug("stfu owed")
+
+	handleErr := func(err error) {
+		l.log.Error(err.Error())
+		l.fail(
+			LinkFailureError{
+				code:             ErrInvalidUpdate,
+				FailureAction:    LinkFailureDisconnect,
+				PermanentFailure: false,
+				Warning:          true,
+			},
+			err.Error(),
+		)
+	}
+
+	localOnRemote := l.channel.NumLocalUpdatesPendingOnRemote()
```

Comment:
Still reading through, but I think we already have methods like `getUnsignedAckedUpdates` and `RemoteUnsignedLocalUpdates`.

Reply:
The thing we are trying to measure is whether all of *our* updates have both been signed and counter-signed. The first of those won't work because we still care about the updates even if they haven't yet been ACK'ed. The second one pulls the updates from disk which I don't think is either necessary or desirable.